src/composables/use-i18n-sync.js

```diff
@@ -0,0 +1,147 @@
+  })
+  const showBanner = ref(false)
+  const bannerLocale = reactive({
+    code: i18n.locale,
+    name: currentLocale.value.name,
+  })
+
+  /**
+   * Updates the i18n locale code and sets the html lang and dir attributes.
+   *
+   * @param {Object} [locale]
+   * @param {string} locale.code - the locale code as used in i18n.
+   * @param {string} locale.lang - value of the HTML lang property.
+   * @param {'rtl'|'ltr'} locale.dir - text direction.
+   * @returns {Promise<void>}
+   */
+  const updateLocale = async (
+    locale = {
+      code: 'en',
+      lang: 'en-US',
+      dir: 'ltr',
+    }
+  ) => {
+    await i18n.setLocale(locale.code)
+    document.documentElement.lang = locale.lang
+    // Always set `dir` property, default to 'ltr'
+    document.documentElement.dir = locale.dir
+  }
+  /**
+   * Show banner inviting to contribute translations if fewer than 90%
+   * of strings are translated for current locale.
+   * Hard-coded to false for default locales: `en` and `en_US`.
+   *
+   * @param {import('../locales/scripts/types').i18nLocaleProps} locale - current locale object
+   * @returns {boolean}
+   */
+  const needsTranslationBanner = (locale) => {
+    if (['en', 'en_US'].includes(locale)) return false
+
+    return locale.translated <= 90
+  }
+
+  const bannerDismissedForLocales = useStorage(
+    'openverse-dismissed-banner-locales',
+    [],
+    {
+      serializer: StorageSerializers.object,
+    }
+  )
+  const shouldHideBanner = computed(() => {
+    return (
+      !showBanner.value ||
+      bannerDismissedForLocales.value.includes(bannerLocale.code)
+    )
+  })
+  const dismissBanner = () => {
+    bannerDismissedForLocales.value = [
+      ...bannerDismissedForLocales.value,
+      bannerLocale.code,
+    ]
+  }
+  const translationLink = computed(
+    () => `${BASE_URL}${bannerLocale.code}/default/`
+  )
+  /**
+   * Handles messages of type `localeSet` received by the `iframe`. Any
+   * other message types will be discarded.
+   *
+   * It sets the i18n locale property, defaulting to `en` if 0 translations are available for the locale,
+   * and shows the translation banner if necessary,
+   *
+   * @param {string} type - the nature of the message received.
+   * @param {Object} value - the data sent with the message.
+   * @param {string} value.locale - wpLocale code of the locale, e.g. 'en_US'.
+   * @param {string} value.lang - the iso code for language
+   * (and sometimes country), e.g. 'en-US'.
+   * Default lang value on wp.org is 'en', and on wp.org/openverse - 'en-US'.
+   * @param {'rtl'|'ltr'} [value.dir] - the locale text direction.
+   */
+  const localeMsgHandler = async ({ data: { type, value } }) => {
+    if (type !== 'localeSet') return
+    // If the locale set by wp.org is 'en_US', not 'en', this is not necessary.
+    const wpLocaleValue = value.locale === 'en' ? 'en_US' : value.locale
+    let locale = i18n.locales.find((item) => item.wpLocale === wpLocaleValue)
```

Comment:
When I test this PR, I'm not able to see the banner anywhere. It seems that `i18n.locales` does not have the `translated` key for each locale. If I manually add `locale.translated = 2` inside of `localeMsgHandler` however, I'm able to see the banner and everything looks great!

<img width="1265" alt="CleanShot 2021-12-09 at 13 53 34@2x" src="https://user-images.githubusercontent.com/6351754/145458065-dffc3c92-f9ba-4cfa-aced-bc01f91755b8.png">

Reply:
Did you run `npm run i18n:create-locales-list` and `npm run i18n:get-translations`? I should've added that to testing instructions. This way, your `locale` lists should get updated with `translated` prop added (both the ones that are used in Vue in i18n.locales, and the ones that have no translations at all).

I wonder if I should:
**A.** Move the dark yellow border to the right for RTL languages (because it looks ugly with the scrollbar that I get when testing with the wp theme) or 
**B.** Remove it completely.

Reply:
Jumping in to say that moving it to the right for RTL languages. I envision that in the long-term future we will work on a different solution for the whole WordPress site.